src/oop/special_class_method/hard/q10/MainTest.java

```diff
@@ -0,0 +1,199 @@
+        assertTrue(testClass.hasField(secondFieldName, String.class),
+                "Your " + secondFieldName + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(secondFieldName, "private"),
+                "Your " + secondFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("First", "Second"),
+                Arguments.of("Science", "Anna"),
+                Arguments.of("Jack", "History"),
+                Arguments.of("Finn", "Fiction")
+
+        );
+    }
+
+    @Test
+    public void awardClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {String.class, String.class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void awardClassHasCorrectConstructor(String value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, String.class},
+                {value2, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectString1ValueMessage =
+                "Your " + className + " constructor does not initialize the " + firstFieldName + " attribute to the correct value.";
+        String incorrectString2ValueMessage =
+                "Your " + className + " constructor does not initialize the " + secondFieldName + " attribute to the correct value.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, firstFieldName),
+                incorrectString1ValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, secondFieldName),
+                incorrectString2ValueMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void awardClassHasCorrectSetAwardRecipientMethod(String initialValue, String setValue) throws Throwable {
+        Object[][] arguments = {
+                {initialValue, String.class},
+                {"", String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, String.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setFirstMethodName, "method does not correctly set the value of the", firstFieldName, "attribute.");
+        Object setMethodOutput = testClass.callMethod(setFirstMethodName, setSizeArguments, setMethodModifiers, classInstance);
```

Comment:
use .hasMethod first and void.type instead of assertNull here and everywhere else